command/agent/command.go

```diff
@@ -295,13 +295,26 @@ func (c *Command) setupAgent(config *Config, logOutput io.Writer, logWriter *log
 		return err
 	}
 
-	rpcListener, err := net.Listen("tcp", rpcAddr.String())
+	if _, ok := rpcAddr.(*net.UnixAddr); ok {
+		// Remove the socket if it exists, or we'll get a bind error
+		_ = os.Remove(rpcAddr.String())
```

Comment:
Hmm, so maybe what we need to do is after the `Listen()`, if we are using a domain socket, defer the removal of it. This way we clean up after ourselves when the agent exits. To me it seems like blindly replacing the socket file could cause unexpected behavior. We probably *should* let `Listen()` error if the socket file we want already exists.